include/cadmium/celldevs/cell/cell.hpp

```diff
@@ -0,0 +1,224 @@
+#include <cadmium/modeling/message_bag.hpp>
+#include <cadmium/celldevs/delayer/delayer_factory.hpp>
+
+
+namespace cadmium::celldevs {
+    /**
+     * Structure that defines cell messages.
+     * @tparam C the used for representing a cell ID.
+     * @tparam S the type used for representing a cell state.
+     */
+    template <typename C, typename S>
+    struct cell_state_message {
+        C cell_id;
+        S state;
+
+        /**
+         * @brief Cell state message constructor
+         * @param cell_id_in ID of the cell that generates the message
+         * @param state_in State to be transmitted by the cell
+         */
+        explicit cell_state_message(C cell_id_in, S state_in): cell_id(cell_id_in), state(state_in) {}
+
+        /**
+         * Operator overloading function for printing cell state messages
+         * @tparam C the used for representing a cell ID.
+         * @tparam S the type used for representing a cell state.
+         * @param os Operating System's string stream
+         * @param msg cell state message
+         * @return Operating System's string stream containing the stringified cell state
+         */
+        friend std::ostream &operator << (std::ostream &os, const cell_state_message<C, S> &msg) {
+            os << msg.cell_id << " ; " << msg.state;
+            return os;
+        }
+    };
+
+
+    /**
+     * @brief input/output port structure for cells.
+     * @tparam C the used for representing a cell ID.
+     * @tparam S the type used for representing a cell state.
+     */
+    template <typename C, typename S>
+    struct cell_ports_def{
+        struct cell_in: public cadmium::in_port<cell_state_message<C, S>> {};
+        struct cell_out : public cadmium::out_port<cell_state_message<C, S>> {};
+    };
+
+    /**
+     * @brief Abstract DEVS atomic model for defining cells in Cell-DEVS scenarios.
+     * @tparam T the type used for representing time in a simulation.
+     * @tparam C the used for representing a cell ID.
+     * @tparam S the type used for representing a cell state.
+     * @tparam V the type used for representing a neighboring cell's vicinity. By default, it is set to integer.
+     * @tparam C_HASH the hash function used for creating unordered maps with cell IDs as keys.
+     *                By default, it uses the one defined in the standard library
+     */
+    template <typename T, typename C, typename S, typename V=int, typename C_HASH=std::hash<C>>
+    class cell {
+    public:
+
+        using input_ports = std::tuple<typename cell_ports_def<C, S>::cell_in>;
+        using output_ports = std::tuple<typename cell_ports_def<C, S>::cell_out>;
+
+        using NV = std::unordered_map<C, V, C_HASH>;
+        using NS = std::unordered_map<C, S, C_HASH>;
+
+
+        C cell_id;                          /// Cell ID
+        std::vector<C> neighbors;           /// Neighboring cells' IDs
+        T clock;                            /// simulation clock
```

Comment:
Is this a real-time implementation?

Reply:
It isn't. However, it keeps track of the simulated time to perform different delayed output functionalities. It is hidden to the user.

Reply:
Can you comment a little longer what is the use? Usually clock is a word very overloaded in simulation and we want to avoid someone confused about it when looking at it back in 6 months.

Reply:
I changed it to simulation_clock and added a longer description